Add more API Kind and SKU names

felixwa opened the pull request:
Add more API Kind and SKU names.
Model API Kind and SKU as string instead of enum. This would give us and users more flexibility.
Add related unit tests.

felixwa committed:
Add more API kinds/skus and model them as string instead of enum

Add more API kinds/skus.
Model them as strings instead of enum.

felixwa committed:
Simplify test code

Simplify test code a little by reusing an existing test helper method.

felixwa committed:
Merge pull request #2 from felixwa/add-more-apis-2

Add more API kinds/skus and model them as string instead of enum

azuresdkci commented:
Can one of the admins verify this patch?

azurecla commented:
Hi __@felixwa__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (felixwa). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

hovsepm commented:
@azuresdkci add to whitelist

cormacpayne commented on src/ResourceManagement/CognitiveServices/Microsoft.Azure.Management.CognitiveServices/Generated/Models/CognitiveServicesAccountCreateParameters.cs:
```diff
@@ -47,11 +47,14 @@ public CognitiveServicesAccountCreateParameters(Sku sku, Kind kind, string locat
         /// </summary>
         [JsonProperty(PropertyName = "kind")]
-        public Kind Kind { get; set; }
+        public string Kind { get; set; }
```
@felixwa changing the type of this property is a breaking change

felixwa replied:
Change the version number as you suggested.

cormacpayne commented on src/ResourceManagement/CognitiveServices/Microsoft.Azure.Management.CognitiveServices/Generated/Models/Sku.cs:
```diff
@@ -29,19 +29,19 @@ public partial class Sku
         /// </summary>
         [JsonProperty(PropertyName = "name")]
-        public SkuName Name { get; set; }
+        public string Name { get; set; }
```
@felixwa changing the type of this property is a breaking change

felixwa replied:
Change the version number as you suggested.

cormacpayne reviewed (changes requested):
@felixwa Hey Felix, I left a couple of comments. Since CognitiveServices is in preview right now, it is OK to make these breaking changes.

A few more comments:

1) Was this code generated from a swagger spec? If so, you will need to specify which version of AutoRest you used, as well as the swagger spec used, in the `generate.cmd` file located [here](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/src/ResourceManagement/CognitiveServices/Microsoft.Azure.Management.CognitiveServices/generate.cmd)

2) You will need to make a version bump because of the changes you are making. Since there are breaking changes, it looks like you would go to 0.2.0-preview. You will need to make the changes in [project.json](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/src/ResourceManagement/CognitiveServices/Microsoft.Azure.Management.CognitiveServices/project.json) and [AssemblyInfo.cs](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/src/ResourceManagement/CognitiveServices/Microsoft.Azure.Management.CognitiveServices/Properties/AssemblyInfo.cs)

felixwa committed:
Merge remote-tracking branch 'refs/remotes/Azure/AutoRest' into AutoRest

felixwa committed:
Merge remote-tracking branch 'refs/remotes/Azure/AutoRest' into AutoRest

felixwa committed:
Reversion to 2.0 as there are breaking changes

Reversion to 2.0 as there are breaking changes.
Set AutoRest version correctly.

felixwa commented:
@cormacpayne , thanks for the comments.

1. I have changed generate.cmd file to reflect the AutoRest version used to generate this SDK. 
2. The Swagger file URL is still that one though, we made similar changes to that file and it is still under review. 
3. Also I changed the version to 2.0 as you suggested.

Please review.

felixwa commented:
@cormacpayne, thanks for the comments, I have changed version of AutoRest in generate.cmd and Assembly version to 2.0 in project.json/AssemblyInfo.cs.

Please review and let me know if you have any other concerns.

-Felix

From: Cormac McCarthy [mailto:notifications@github.com]
Sent: Thursday, December 22, 2016 1:54 AM
To: Azure/azure-sdk-for-net <azure-sdk-for-net@noreply.github.com>
Cc: Felix Wang <felixwa@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Azure/azure-sdk-for-net] Add more API Kind and SKU names (#2657)


@cormacpayne requested changes on this pull request.

@felixwa<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Ffelixwa&data=02%7C01%7Cfelixwa%40microsoft.com%7Cb6fc4fb0193e4fdf231808d429ca6193%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636179396581370678&sdata=qlrQ6wWMmxkU2naJpQBeYy8PHirBGT%2FzbjPnVLPC70w%3D&reserved=0> Hey Felix, I left a couple of comments. Since CognitiveServices is in preview right now, it is OK to make these breaking changes.

A few more comments:

  1.  Was this code generated from a swagger spec? If so, you will need to specify which version of AutoRest you used, as well as the swagger spec used, in the generate.cmd file located here<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-sdk-for-net%2Fblob%2FAutoRest%2Fsrc%2FResourceManagement%2FCognitiveServices%2FMicrosoft.Azure.Management.CognitiveServices%2Fgenerate.cmd&data=02%7C01%7Cfelixwa%40microsoft.com%7Cb6fc4fb0193e4fdf231808d429ca6193%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636179396581370678&sdata=sRwTpYo7br3YSAVEu3%2FO3Ad0wvy2Q2gF83%2BCyTBfpZI%3D&reserved=0>
  2.  You will need to make a version bump because of the changes you are making. Since there are breaking changes, it looks like you would go to 0.2.0-preview. You will need to make the changes in project.json<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-sdk-for-net%2Fblob%2FAutoRest%2Fsrc%2FResourceManagement%2FCognitiveServices%2FMicrosoft.Azure.Management.CognitiveServices%2Fproject.json&data=02%7C01%7Cfelixwa%40microsoft.com%7Cb6fc4fb0193e4fdf231808d429ca6193%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636179396581370678&sdata=mRXOunRdgAvN%2FTepHKqH1eD%2FePa5aA31J53CDcgkde8%3D&reserved=0> and AssemblyInfo.cs<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-sdk-for-net%2Fblob%2FAutoRest%2Fsrc%2FResourceManagement%2FCognitiveServices%2FMicrosoft.Azure.Management.CognitiveServices%2FProperties%2FAssemblyInfo.cs&data=02%7C01%7Cfelixwa%40microsoft.com%7Cb6fc4fb0193e4fdf231808d429ca6193%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636179396581370678&sdata=cjiU%2BPqEfVbOjn2yRkscFE4xdY9NadCSxsCbkojBRHQ%3D&reserved=0>

________________________________

In src/ResourceManagement/CognitiveServices/Microsoft.Azure.Management.CognitiveServices/Generated/Models/CognitiveServicesAccountCreateParameters.cs<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-sdk-for-net%2Fpull%2F2657%23pullrequestreview-14015174&data=02%7C01%7Cfelixwa%40microsoft.com%7Cb6fc4fb0193e4fdf231808d429ca6193%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636179396581370678&sdata=xSMG0WdA9Mz8%2FRx22uwVe%2BY8zU5p%2BAo3peGdEijNEPM%3D&reserved=0>:

>          /// </summary>

         [JsonProperty(PropertyName = "kind")]

-        public Kind Kind { get; set; }

+        public string Kind { get; set; }

@felixwa<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Ffelixwa&data=02%7C01%7Cfelixwa%40microsoft.com%7Cb6fc4fb0193e4fdf231808d429ca6193%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636179396581370678&sdata=qlrQ6wWMmxkU2naJpQBeYy8PHirBGT%2FzbjPnVLPC70w%3D&reserved=0> changing the type of this property is a breaking change

________________________________

In src/ResourceManagement/CognitiveServices/Microsoft.Azure.Management.CognitiveServices/Generated/Models/Sku.cs<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-sdk-for-net%2Fpull%2F2657%23pullrequestreview-14015174&data=02%7C01%7Cfelixwa%40microsoft.com%7Cb6fc4fb0193e4fdf231808d429ca6193%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636179396581370678&sdata=xSMG0WdA9Mz8%2FRx22uwVe%2BY8zU5p%2BAo3peGdEijNEPM%3D&reserved=0>:

>          /// </summary>

         [JsonProperty(PropertyName = "name")]

-        public SkuName Name { get; set; }

+        public string Name { get; set; }

@felixwa<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Ffelixwa&data=02%7C01%7Cfelixwa%40microsoft.com%7Cb6fc4fb0193e4fdf231808d429ca6193%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636179396581370678&sdata=qlrQ6wWMmxkU2naJpQBeYy8PHirBGT%2FzbjPnVLPC70w%3D&reserved=0> changing the type of this property is a breaking change

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-sdk-for-net%2Fpull%2F2657%23pullrequestreview-14015174&data=02%7C01%7Cfelixwa%40microsoft.com%7Cb6fc4fb0193e4fdf231808d429ca6193%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636179396581370678&sdata=xSMG0WdA9Mz8%2FRx22uwVe%2BY8zU5p%2BAo3peGdEijNEPM%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAPSimnuBNr8DeLU9wepz5JNAB9rxg91aks5rKWfFgaJpZM4LQdzU&data=02%7C01%7Cfelixwa%40microsoft.com%7Cb6fc4fb0193e4fdf231808d429ca6193%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636179396581370678&sdata=AGM0TDp07PlOLojJ9O0fcmvVMYIVuCk%2F3voaulwqV50%3D&reserved=0>.

cormacpayne commented:
@felixwa Hey Felix, since you used version 0.17.x of AutoRest from a few months ago, would you mind using one of the latest nightly builds to regenerate your code? We are up to version 1.0.0, and you can find out how to access it [here](https://github.com/Azure/autorest#getting-autorest)

felixwa committed:
Use AutoRest 1.0.0 to regenerate all model files.

Now use AutoRest 1.0.0 to regenerate all model files.

felixwa commented:
@cormacpayne , now I have used AutoRest 1.0.0 20170101 to regenerate all model files.

felixwa commented:
@cormacpayne, ping...API Spec side had accepted and merged my change, can you review and take a look at this one at the earliest? So that I can continue to modify PS for Cognitive Services.
